tests/epics/signal/test_signals.py

```diff
@@ -43,41 +43,98 @@
 )
 from ophyd_async.epics.signal._signal import _epics_signal_backend  # noqa: PLC2701
 
-RECORDS = str(Path(__file__).parent / "test_records.db")
+CA_PVA_RECORDS = str(Path(__file__).parent / "test_records.db")
+PVA_RECORDS = str(Path(__file__).parent / "test_records_pva.db")
 PV_PREFIX = "".join(random.choice(string.ascii_lowercase) for _ in range(12))
 
+Protocol = Literal["ca", "pva"]
+
+
+PARAMETERISE_PROTOCOLS = pytest.mark.parametrize("protocol", get_args(Protocol))
+
+
+class DeviceTestRecordsGroup(Device):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self._all_signals = {"ca": {}, "pva": {}}
+        for protocol in get_args(Protocol):
+            self.__create_signal(protocol, int, "int")
+            self.__create_signal(protocol, float, "float")
+            self.__create_signal(protocol, str, "str")
+            self.__create_signal(protocol, MyEnum, "enum")
+            self.__create_signal(protocol, MyEnum, "enum2")
+            self.__create_signal(protocol, bool, "bool")
+            self.__create_signal(protocol, bool, "bool_unnamed")
+            self.__create_signal(protocol, int, "partialint")
+            self.__create_signal(protocol, int, "lessint")
+            self.__create_signal(protocol, Array1D[np.uint8], "uint8a")
+            self.__create_signal(protocol, Array1D[np.int16], "int16a")
+            self.__create_signal(protocol, Array1D[np.int32], "int32a")
+            self.__create_signal(protocol, Array1D[np.float32], "float32a")
+            self.__create_signal(protocol, Array1D[np.float64], "float64a")
+            self.__create_signal(protocol, Sequence[str], "stra")
+        self.__create_signal("pva", Array1D[np.int8], "int8a")
+        self.__create_signal("pva", Array1D[np.uint16], "uint16a")
+        self.__create_signal("pva", Array1D[np.uint32], "uint32a")
+        self.__create_signal("pva", Array1D[np.int64], "int64a")
+        self.__create_signal("pva", Array1D[np.uint64], "uint64a")
+        self.__create_signal("pva", MyTable, "table")
+        self.__create_signal("pva", Array1D[np.int64], "ntndarray:data")
+
+    def __create_signal(
+        self, protocol: Protocol, dtype: type[SignalDatatypeT], name: str
+    ):
+        signal = epics_signal_rw(dtype, f"{protocol}://{self._name}:{protocol}:{name}")
+        setattr(self, f"{protocol}_{name}".replace(":", "_"), signal)
+        self._all_signals[protocol][name] = signal
+
+    def get_signal(self, protocol: Protocol, name: str):
+        return self._all_signals[protocol][name]
```

Comment:
Actually, if we base this on https://github.com/bluesky/ophyd-async/pull/598 then we could write this like:
```python
from typing import Annotated as A
from ophyd_async.epics.core import EpicsDevice, PvSuffix

class CaAndPvaTestDevice(EpicsDevice):
    int: A[SignalRW[int], PvSuffix("int")]
    ...
    stra: A[Sequence[str], PvSuffix("stra")]
```

Then we can add the PVA only ones in a second device:
```python
class PvaOnlyTestDevice(EpicsDevice):
    int8a: A[Array1D[np.int8], PvSuffix("int8a")]
    ...
    ntndarray: A[np.ndarray, PvSuffix("ntndarray")]
```